### PR TITLE
[WIP] Preparing for a conversation

### PR DESCRIPTION
**Pull request by jonmagic**
This is a WIP, the conversation hasn't been completely coded and what I have so far is ripe for refactoring.

The final goal is being able to impersonate the cd changer during the conversation that happens between the stereo and cd changer after being powered on.

### CODE REVIEW

File: src/brain.ino [hunk 1]
```diff
@@ -27,50 +28,90 @@ void send_message(char *msg) {
       write_one();
     }
   }
+
+  Serial.print("> ");
+  Serial.println(msg);
```
Comment (jch) [inline src/brain.ino:33]: <3 the `> <` for readability

File: src/brain.ino [hunk 2]
```diff
@@ -27,50 +28,90 @@ void send_message(char *msg) {
...
         } else if(pulse_width > 500 && pulse_width < 700) {
-          message.concat("1");
+          // record a 1
+          msg.concat("1");
+          // set pulse_end to the current microseconds timestamp
           pulse_end = micros();
+          // update expected_micros_to_next_pulse to the largest expected value
           expected_micros_to_next_pulse = 510;
+        // if the pulse is not a 0 or 1 send a debug message to Serial so we can
+        // go WTF
```
Comment (jch) [inline src/brain.ino:87]: lol

File: src/brain.ino [hunk 3]
```diff
@@ -79,36 +120,132 @@ void setup() {
...
+
+  // listen for head1 until we hear it
+  char head1[] = "1101010100";
+  while(unmatched) {
+    response = listen_for_message();
+
+    if(response == head1) {
+      unmatched = false;
+    }
+  }
```
Comment (jch) [inline src/brain.ino:133]: I think this is a good first start. If we don't get the message, or if we only get head1, but not head2, we'd see it hang here :+1:

File: src/brain.ino [hunk 4]
```diff
@@ -79,36 +120,132 @@ void setup() {
...
+  while(unmatched) {
+    response = listen_for_message();
+
+    if(response == head4) {
+      unmatched = false;
+    }
+  }
+  unmatched = true;
+
+  // listen for head5 until we hear it
```
Comment (jch) [inline src/brain.ino:172]: So there are 5 messages from the radio before the CD acks?

File: src/brain.ino [hunk 5]
```diff
@@ -79,36 +120,132 @@ void setup() {
...
+  while(unmatched) {
+    response = listen_for_message();
+
+    if(response == head6) {
+      unmatched = false;
+    }
+  }
+  unmatched = true;
+
+  // say hello again because why not
```
Comment (jch) [inline src/brain.ino:217]: :grin:

File: src/brain.ino (hunk 4 above)
Comment (jonmagic) [inline src/brain.ino:172]: Looks like it.

**Comment by jch** [comment]
This is so awesome. Love the comments.

**Comment by jonmagic** [comment]
Closing this for now as it's bugging me on my /pulls page :grin:

**Pull request closed without merging**